build.go

```diff
@@ -19,13 +19,15 @@ import (
 )
 
 var verbose bool
+var configureOptions string
 var autopointPath string
 var folders = []string{"openssl", "libevent", "zlib", "xz", "tor"}
 var absCurrDir = getAbsCurrDir()
 var numJobs = fmt.Sprintf("-j%d", runtime.NumCPU())
 
 func main() {
 	flag.BoolVar(&verbose, "verbose", false, "Whether to show command output")
+	flag.StringVar(&configureOptions, "configure-options", "", "Configure options")
```

Comment:
Hrmm, not sure it makes sense to pass the same configure options to every make target. What's an example of this value? Can we address the reasoning behind needing this more specifically instead of this generic option? The goal is to not need much configuration to run this. If there's another mode we need to run this in or another use case we should support that we don't, we should add an option for that case specifically IMO instead of these super flexible but less clear options.

Reply:
`-configure-options` is for each build command, such as `build-all`, `build-openssl`, `build-libevent` etc. If we use `-configure-options` for `build-all` then it will used for each subcommand.

I'm using bine and tor-static in [GostX](https://github.com/lkebin/GostX/blob/ca402c73d623adf7690fd5df523efbb28f390061/go/Makefile#L51), it needs cross-compilation in Github CI. There's my example in GostX (Build env is Darwin x86_64):

```
CFLAGS="-target arm64-apple-macos11" $(GO) run build.go -verbose -configure-options="darwin64-arm64-cc" build-openssl 
CFLAGS="-target arm64-apple-macos11" $(GO) run build.go -verbose -configure-options="--host=arm64-apple-macos11" build-libevent 
CFLAGS="-target arm64-apple-macos11" $(GO) run build.go -verbose build-zlib 
CFLAGS="-target arm64-apple-macos11" $(GO) run build.go -verbose -configure-options="--host=arm64-apple-macos11" build-xz 
CFLAGS="-target arm64-apple-macos11" $(GO) run build.go -verbose -configure-options="--host=arm64-apple-macos11 --disable-tool-name-check" build-tor
```

For cross-compilation, libevent / xz / tor need to set `--host` option, and tor also need `--disable-tool-name-check` option on macOS. openssl needs a specified OS compiler.

I considered adding `-host` flag, but it is not enough for tor and openssl. 

All these changes' aim is to support cross-compilation.

Reply:
Most people I suspect run `build-all` so the arg doesn't make sense there, but we can leave it here.

But the goal of this project was to make compilation easy. Once you start adding general purpose knobs it becomes less easy. I see you made an update below to set these things based on things you determine from the environment. I think that's much better than general configure options.

For example, what would you need to do to make sure the way you're compiling tor-static for your project works with `build-all`? Can we make those changes?

Reply:
Replace to `-host` flag for cross-compilation, also some workaround for tor and openssl. It should be easier than `-configure-options`.

It is working with `build-all` command, also tested in my project [GostX](https://github.com/lkebin/GostX/blob/329bcf2d73fd828afe36b044dcde70529205b5a6/go/Makefile#L51)

```
CFLAGS="-target arm64-apple-macos11" $(GO) run build.go -verbose -host="arm64-apple-macos11" build-all
```

A `boolean` flag may not be enough for cross-compilation, it needs specified compiler info. How about `-host` or `-cross-compilation-host` ?

Reply:
I think `-host` works perfectly. May want to document w/ a note in README, but LGTM.

Reply:
I did test on macOS only, not sure if it works for other platforms. We could merge it and add documentation soon